### PR TITLE
fix rust doc comments

### PR DESCRIPTION
**Pull request by Luap99**
Rust doc comments need three slashes[1]. Using just `//` makes rust think they are a normal comment which should not be included in the API docs. This is important if you use function highlighting in my IDE for example.

[1] https://doc.rust-lang.org/reference/comments.html

### CODE REVIEW

**Comment by Luap99** [comment]
@mheon @flouthoc @baude PTAL. I know I make this mistake as well but try to remember that you have to use `///` to comment functions or types in rust  unless you really do not want them to appear in the docs.

**Review by flouthoc — APPROVED** [review]
LGTM

File: src/error/mod.rs [hunk 1]
```diff
@@ -81,7 +81,7 @@ pub enum NetavarkError {
     List(NetavarkErrorList),
 }
 
-// Internal struct for JSON output
+/// Internal struct for JSON output
```
Comment (mheon) [inline src/error/mod.rs:84]: Do we need this? It's not `pub`?

**Comment by mheon** [comment]
General question: should we bother with rustdoc on non-public functions? It seems like we should focus on rustdoc for things that could actually be used outside of the package.

**Comment by Luap99** [comment]
> General question: should we bother with rustdoc on non-public functions? It seems like we should focus on rustdoc for things that could actually be used outside of the package.

Well it should still show up as hint when I hover the function name in my IDE. I assume that if you render the docs as web page it will omit non public types/functions,

**Comment by flouthoc** [comment]
> General question: should we bother with rustdoc on non-public functions? It seems like we should focus on rustdoc for things that could actually be used outside of the package.

@mheon Lot of editior extensions use those to show helpers so I guess we can keep it as it is.

**Pull request merged**